buildpacks/java/java/detect.go

```diff
@@ -53,9 +53,10 @@ func (d Detect) Detect(context libcnb.DetectContext) (libcnb.DetectResult, error
 			{
 				Name: "java-function",
 				Metadata: map[string]interface{}{
-					"launch":            true,
-					"func_yaml_envs":    funcYaml.Envs,
-					"func_yaml_options": funcYaml.Options,
+					"launch":                  true,
+					"func_yaml_envs":          funcYaml.Envs,
+					"func_yaml_function_name": funcYaml.Name,
```

Comment:
I think I would prefer it to just be `func_yaml_name` to match the other keys better, as you store it in the struct as `Name` and the getter is `getName`.